db/migrate/20230605133804_remove_master_variants.rb

```diff
@@ -0,0 +1,40 @@
+class RemoveMasterVariants < ActiveRecord::Migration[7.0]
```

Comment:
It's the main point of the pull request. Taking this out would mean taking a lot of other changes out. We would still need to filter by `is_master: false` by default, for example.

In general, yes, I would have split this PR into smaller chunks and this would have come later. But this PR was built as a whole and taking things out could be a lot more work, including testing that Matt has already done on the whole thing.

So I'm okay with this. It's slightly risky but I think that we can handle it.
Hm, this may actually cause some downtime as well. Maybe we need a rollout plan for this PR.

Reply:
> Taking this out would mean taking a lot of other changes out. We would still need to filter by `is_master: false` by default, for example.

Oh yeah, good point. I have to admit my brain was pretty fried by this point ;)

> Hm, this may actually cause some downtime as well. Maybe we need a rollout plan for this PR.

Yeah, if records are deleted before Puma (and Sidekiq) reload the new code, we will probably have some errors. Maybe not many of much consequence.. but would need to think through that before making that judgement.

If it can be deployed at a quiet time for each instance's timezone, it might be fine. 
Or otherwise we can schedule a downtime window. Hmm.. deployments can take quite some time (I haven't timed lately but it could be up to 20 mins). That's not great. Perhaps we can update the deploy playbook with a downtime option that only brings it down for the minimum time needed.

So maybe we should test out a live deployment and hope for the best?

Reply:
> Luckily we already have a playbook which does exactly that

Great, I didn't realise that! That will result in downtime for the entire deploy process. I haven't reviewed the deploy process, but was wondering if we can minimise the downtime by limiting it only to the core part of the process. 
I've created an issue, but I don't think we necessary need to do it this time.